### PR TITLE
[New Theme] Rename to `Next`

### PR DESCRIPTION
**Pull request by AMoo-Miki**
### Description
This renames the file and references to the old name.

Also, certain comments that talk about the old beta theme's specifications or limitations are reworded or removed. There are 2 instances that more information is needed before removing some logic.

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [X] All tests pass
  - [ ] `yarn lint`
  - [ ] `yarn test-unit`
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/oui/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src-docs/src/theme_next_dark.scss [hunk 1]
```diff
@@ -9,10 +9,7 @@
  * GitHub history for details.
  */
 
-// sass-lint:disable no-url-domains, no-url-protocols
-@import url('https://fonts.googleapis.com/css2?family=Inter:wght@300;400;500;600;700&family=Roboto+Mono:ital,wght@0,400;0,700;1,400;1,700&display=swap');
```
Comment (BSFishy) [inline src-docs/src/theme_next_dark.scss:13]: Should this be replaced with Source Sans & Source Code fonts?

File: src-docs/src/theme_next_light.scss [hunk 2]
```diff
@@ -9,10 +9,7 @@
  * GitHub history for details.
  */
 
-// sass-lint:disable no-url-domains, no-url-protocols
-@import url('https://fonts.googleapis.com/css2?family=Inter:wght@300;400;500;600;700&family=Roboto+Mono:ital,wght@0,400;0,700;1,400;1,700&display=swap');
```
Comment (BSFishy) [inline src-docs/src/theme_next_light.scss:13]: Same as previous comment

File: src-docs/src/theme_next_dark.scss (hunk 1 above)
Comment (AMoo-Miki) [reply src-docs/src/theme_next_dark.scss:13]: Yep! i knew i forgot something. Thanks.

File: src/components/page/_restrict_width.ts [hunk 3]
```diff
@@ -44,7 +44,7 @@ import { CSSProperties } from 'react';
 export type _OuiPageRestrictWidth = {
   /**
    * Sets the max-width of the page,
-   * set to `true` to use the default size of `1000px (1200 for Cascadia)`,
```
Comment (joshuarrrr) [inline src/components/page/_restrict_width.ts:47]: So the default size is 1000px for all themes in the OUI project as of this PR?

**Review by joshuarrrr — COMMENTED** [review]
Just some minor questions, but looks good.

File: src-docs/src/views/panel/panel_shadow.js [hunk 4]
```diff
@@ -9,15 +9,11 @@
...
  */
 
-import React, { useContext } from 'react';
+import React from 'react';
 
 import { OuiPanel, OuiCode, OuiSpacer } from '../../../../src/components';
-import { ThemeContext } from '../../components';
 
 export default () => {
-  const themeContext = useContext(ThemeContext);
```
[the remark this answers is not in the document]
Comment (AMoo-Miki) [reply src-docs/src/views/panel/panel_shadow.js:18]: This guy is responsible for switching between light and dark themes. I donno how we switch from default to "Next" in the docs but will figure it out.

File: src-docs/src/theme_next_dark.scss (hunk 1 above)
Comment (BSFishy) [reply src-docs/src/theme_next_dark.scss:13]: To note: here, this is to enable the fonts on the docsite, not to package the fonts with OUI. OUI doesn't package any fonts out of the box, that is up to the implementor

File: src-docs/src/views/panel/panel_shadow.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (BSFishy) [reply src-docs/src/views/panel/panel_shadow.js:18]: To add theme to docsite: https://github.com/opensearch-project/oui/compare/1.1.0...pink-elephant#diff-235116db4d88026739c19ef26262c8b53df2d35cdd4f2ff62d8a8d42bf7ed975R36 and https://github.com/opensearch-project/oui/compare/1.1.0...pink-elephant#diff-393e09a1e9d03ae7b852765c1352fe66181b65d531e054ee0c3f7d4c6cc923a5R45-R48

[the remark this answers is not in the document]
Comment (BSFishy) [reply src-docs/src/views/panel/panel_shadow.js:18]: Also, theme context is responsible for switching between themes in general on docsite, not just light/dark

File: src-docs/src/views/panel/panel_example.js [hunk 5]
```diff
@@ -168,10 +168,8 @@ export const PanelExample = {
             title="Certain allowed combinations of shadow, border, and color depend on the current theme.">
             <p>
               For instance, only plain or transparent panels can have a border
-              and/or shadow. The Cascadia theme doesn&apos;t allow combining the{' '}
-              <OuiCode>hasBorder</OuiCode> option with{' '}
-              <OuiCode>hasShadow</OuiCode>. The default theme only allows
-              removing the border if both <OuiCode>hasShadow</OuiCode> and{' '}
+              and/or shadow. The default theme only allows removing the border
+              if both <OuiCode>hasShadow</OuiCode> and{' '}
```
[the remark this answers is not in the document]
Comment (AMoo-Miki) [reply src-docs/src/views/panel/panel_example.js:172]: cas* was a temporary reference to a temporary and incomplete theme; that was reset to default in #816. The new colors are just named "Next"

File: src-docs/src/views/panel/panel_shadow.js [hunk 6]
```diff
@@ -9,15 +9,11 @@
  * GitHub history for details.
  */
 
-import React, { useContext } from 'react';
+import React from 'react';
 
 import { OuiPanel, OuiCode, OuiSpacer } from '../../../../src/components';
-import { ThemeContext } from '../../components';
 
 export default () => {
```
[the remark this answers is not in the document]
Comment (AMoo-Miki) [reply src-docs/src/views/panel/panel_shadow.js:16]: The previous beta theme had limitations and `ThemeContext` was used to apply rules when dealing with those limitations. Next is evolved out of default and as such, has none of those limitations so. there was no need to know the `ThemeContext` in this file.

File: src-docs/src/views/panel/panel_shadow.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (AMoo-Miki) [reply src-docs/src/views/panel/panel_shadow.js:18]: Matt is right; for OUI, light, dark, next-light, and next-dark are individual themes.

**Review by BSFishy — APPROVED** [review]

**Review by bandinib-amzn — APPROVED** [review]
LGTM! Thank you for the change.

**Pull request merged**